### PR TITLE
Use Detox for greybox e2e testing

### PR DESCRIPTION
**Pull request by markholland**
Tried out Detox for on device e2e testing, only setup for iOS but Android is supported.

- https://github.com/wix/detox
- https://hackernoon.com/detox-gray-box-end-to-end-testing-framework-for-mobile-apps-196ccd9564ce

- [x] Make sure build is able to load from contentful i.e. write a test that interacts with an event.

### CODE REVIEW

File: e2e/init.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import detox from "detox";
+import { detox as config } from "../package.json";
+
+// Set the default test timeout of 120s
+jest.setTimeout(120000);
```
Comment (x4d3) [inline e2e/init.js]: out of curiosity, why 120 seconds ?

**Comment by robbiemccorkell** [comment]
Detox looks like a pretty nice tool, hadn't seen it before. I like the fact it's all promise based. I hate it when e2e testing libraries try to be too clever.

File: e2e/init.js (hunk 1 above)
Comment (markholland) [reply e2e/init.js]: Abitrary, e2e timeouts are usually quite high to allow for network latency.

**Comment by markholland** [comment]
I'll clean this up today and get Android running as well as I think some basic e2e tests will be helpful.

File: e2e/init.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import detox from "detox";
+import { detox as config } from "../package.json";
```
Comment (charypar) [inline e2e/init.js]: What's the benefit of having detox config in `package.json`?

Comment (markholland) [reply e2e/init.js]: I could move it to its own config file, I'd put it in package.json because the jest config was already there.

**Comment by markholland** [comment]
Considering master already has a regression, tapping on events does nothing, how do we feel about merging this with it only running on iOS for now? I think very basic manual testing can be quickly automated and kept up to date. The run time is currently similar to the iOS build so shouldn't cause any disruption.

**Comment by robbiemccorkell** [comment]
Wow, that's a big regression I didn't realise. Yea sounds good, I think since we're much lighter on manual testing for this project, and automated golden path is even more valuable.

**Pull request merged**